### DISCUSSION TITLE
code search is inaccurate and does not show all exact matches

### DISCUSSION THREAD

**Discussion by braedontask** [community/community#63335] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

**issue**
when searching in my company's repository for an exact string (in my case: `ExternalEntityIngestionService` in `get-wrecked/medal-backend`) it only delivers 2 of 3 results. ironically, it excludes the definition of this class from the search. 

i presume this might be bug in indexing but ive tried searching with the `repo:get-wrecked/medal-backend` and the repository isnt new (as other discussion items mentioned). hoping this isnt widespread :)

**emotional context**
after leaving meta im highly dependent on strong code search for due diligence on auditing backend callsites. its pretty frightening to get an inconsistent result on code search. i dont want to live in fear.

**Comment by look** [comment, marked as answer]
Apologies for the frustration. I looked into this and I believe the file is being excluded because one of its path components (`/external/`) triggers our vendored code detection. Unfortunately, there's no work around right now, but I raised the issue to the team for feedback. We're hoping to index all code (or at least, more of it) in the future.

**Reply by braedontask** [reply]
thanks for the response! that information is certainly better than not knowing why the results are inconsistent 👍🏾 

after reading this [post](https://github.blog/2023-02-06-the-technology-behind-githubs-new-code-search/) on the new search engine... i am curious how much memory is saved by not indexing vendored code?

**Reply by Dunedan** [reply]
@look We have the same issue in one of our repository with results from a sub-directory called `vendor` (which actually doesn't contain vendored code) not being shown in the search results.

If Githubs "vendored code detection" just relies on matching of the path, that's obviously leading to lots of false positives being excluded from search.

**Comment by thejasatl** [comment]
(/external/) triggers our vendored code detection

**Reply by deni-indotruck** [reply]
fixit

**Reply by nuritu** [reply]
got it

**Comment by JaniVilen** [comment]
Code search inaccuracies can result from various factors. Inefficient indexing, complex code syntax, or search algorithms can cause issues. To improve results, use precise [keywords](https://cricketagony.com/) and filters. Consider using alternative search engines, tools, or IDE extensions designed for code searching to enhance accuracy and find exact matches effectively.

**Comment by mcobbice** [comment]
What is going on here? Why does github not find any results when I have literally copy-pasted a search term from a repo?

https://github.com/user-attachments/assets/78ce2fba-8161-44e2-831b-9c70aba86a1a

**Reply by look** [reply]
Hi @mcobbice. Sorry for the confusion, but GitHub code search only indexes the default branch (see the [limitations documentation](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations)). You're looking at a tag. I think on the main branch that function is now `zbx_dc_sync_configuration`.

**Reply by mcobbice** [reply]
Ah I see, that makes sense. Thank you for the explanation

**Reply by Brain2000** [reply]
I was having the same problem searching the default branch. I found the URL was magically appending "&type=issues". After removing that, it changed it to "&type=code" and it worked. There seems to be a bug here where it is defaulting the type to "issues" with no indicator.
